src/http.js

```diff
@@ -170,4 +171,13 @@ export default class HTTP {
     }
     this.events.emit("backoff", backoffMs);
   }
+
+  _checkForRetryAfterHeader(status, headers) {
+    let retryAfter = headers.get("Retry-After");
+    if (!retryAfter) {
+      return;
+    }
+    retryAfter = parseInt(retryAfter, 10);
+    this.events.emit("retry-after", retryAfter);
```

Comment:
Yeah, I know what you mean... but damn it's ugly :(

We didn't have to do it for backoff because it provides a date instead of a delay. We should do the same for retry-after IMO.